ERCS/erc-7582.md

```diff
@@ -0,0 +1,100 @@
+
+In all of the above validation steps, the validator contract MUST respect the ERC-4337 Entry Point conventions. Note, that while validator key data might be included elsewhere in a `UserOperation` to achieve similar contract account modularity, for example, by packing this data into the `signature` field, this proposal opts to repurpose `nonce` for this pointer to minimize calldata costs and to benefit from the EntryPoint's `getNonce` accounting, as well as the discoverability of user plugin interactions in the `UserOperationEvent` which exposes `nonce` but not other userOp data.
+
+ERC-4337 references:
+
+[UserOperation](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/interfaces/UserOperation.sol):
+
+```solidity
+    struct UserOperation {
+        address sender; /*contract account*/
+        uint256 nonce; /*validator data*/
+        bytes initCode;
+        bytes callData;
+        uint256 callGasLimit;
+        uint256 verificationGasLimit;
+        uint256 preVerificationGas;
+        uint256 maxFeePerGas;
+        uint256 maxPriorityFeePerGas;
+        bytes paymasterAndData;
+        bytes signature;
+    }
+```
+
+[`IAccount`](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/interfaces/IAccount.sol):
+
+```solidity
+    function validateUserOp(
+        UserOperation calldata userOp, /*userOp.nonce: validator data*/
+        bytes32 userOpHash,
+        uint256 missingAccountFunds
+    ) external returns (uint256 validationData);
+```
+
+[`IEntryPoint`](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/interfaces/IEntryPoint.sol):
+
+```solidity
+    event UserOperationEvent(
+        bytes32 indexed userOpHash,
+        address indexed sender,
+        address indexed paymaster,
+        uint256 nonce, /*validator data*/
+        bool success,
+        uint256 actualGasCost,
+        uint256 actualGasUsed
+    );
+```
+
+[NonceManager](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/core/NonceManager.sol)
+
+```solidity
+    function getNonce(address sender, uint192 key /*validator data*/) external view returns (uint256 nonce);
+```
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Reference Implementation
+
+See `https://github.com/NaniDAO/7582-account`
```

Comment:
You'll either need to add your reference implementation into your `assets/` directory, or remove this section entirely:

```suggestion
```

Reply:
Can the reference implementation in `assets/` directory link to external directories? In this case to 4337 BaseAccount implementation.

Reply:
We prefer self contained reference implementations. You can do a hand-wavy reference to another project (`import { IERC20 } from '@openzeppelin/contracts/token/ERC20/IERC20.sol';` is pretty common), but avoid any explicit URLs.